syft/pkg/cataloger/golang/parse_go_bin.go

```diff
@@ -0,0 +1,125 @@
+package golang
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/anchore/syft/syft/pkg"
+)
+
+const packageIdentifier = "dep"
+
+func parseGoBin(path string, reader io.ReadCloser) ([]pkg.Package, error) {
+	pkgsSlice := make([]pkg.Package, 0)
+
+	// Identify if bin was compiled by go
+	x, err := openExe(reader)
+	if err != nil {
+		return pkgsSlice, err
+	}
+
+	_, mod := findVers(x)
+	fields := strings.Fields(mod)
```

Comment:
Splitting off the parsing of the bytes from the build section into a separate function that can be unit tested would be ideal here (both for reading of this function and for an extra opportunity to test this interface)